ccm_web/server/src/api/api.create.section.handler.ts

```diff
@@ -0,0 +1,66 @@
+import CanvasRequestor from '@kth/canvas-api'
+import baseLogger from '../logger'
+import { CreateSectionTempDataStore, CanvasSectionBase, CreateSectionsAPIErrorData, APIErrorData, isAPICreateSectionErrorData } from './api.interfaces'
+import { handleAPIError } from './api.utils'
+
+const logger = baseLogger.child({ filePath: __filename })
+
+export class CreateSectionApiHandler {
+  requestor: CanvasRequestor
+  sections: string[]
+  courseId: number
+
+  constructor (requestor: CanvasRequestor, sections: string[], courseId: number) {
+    this.requestor = requestor
+    this.sections = sections
+    this.courseId = courseId
+  }
+
+  async apiCreateSection (sectionName: string): Promise<CanvasSectionBase | CreateSectionsAPIErrorData> {
+    try {
+      const endpoint = `courses/${this.courseId}/sections/ding`
+      const method = 'POST'
+      const requestBody = { course_section: { name: sectionName } }
+      logger.debug(`Sendings request to Canvas - Endpoint: ${endpoint}; Method: ${method}; Body: ${JSON.stringify(requestBody)}`)
+      const response = await this.requestor.requestUrl<CanvasSectionBase>(endpoint, method, requestBody)
+      const { id, name } = response.body
+      return { id, name }
+    } catch (error) {
+      const errResponse: APIErrorData = handleAPIError(error)
+      return { statusCode: errResponse.statusCode, errors: [{ message: errResponse.message, failedInput: sectionName }] }
+    }
+  }
+
+  makeReturnResponseCreateSections (sectionsReturnRes: Array<CreateSectionsAPIErrorData | CanvasSectionBase>): CanvasSectionBase[] | CreateSectionsAPIErrorData {
+    const sectionsDataStore: CreateSectionTempDataStore = { allSuccess: [], statusCode: [], errors: [] }
+    for (const section of sectionsReturnRes) {
+      if (isAPICreateSectionErrorData(section)) {
+        const { statusCode, errors } = section
+        sectionsDataStore.errors.push(errors[0])
```

Comment:
Do you see any possible issues with using `sectionDataStore.errors.push(...errors)` instead? I guess I feel minorly uncomfortable with the indexing (emphasis minorly).

Reply:
No, this works nicely.